test/unit/lib/migration-chunks/validation/field/change-id.spec.js

```diff
@@ -0,0 +1,219 @@
+'use strict';
+
+const { expect } = require('chai');
+const Bluebird = require('bluebird');
+
+const migrationPlan = require('../../../../../../lib/migration-chunks');
+const migrationSteps = require('../../../../../../lib/migration-steps');
+const validatePlan = require('../../../../../../lib/migration-chunks/validation');
+
+const stripCallsite = require('../../../../../helpers/strip-callsite');
+const stripCallsites = (plan) => plan.map((chunk) => chunk.map(stripCallsite));
+
+describe('chunks validation when changing field ids', () => {
+  it('returns an error when referring to a field by its old id in the same migration', Bluebird.coroutine(function * () {
+    const steps = yield migrationSteps(function up (migration) {
+      const book = migration.editContentType('book');
+      book.changeFieldId('title', 'newTitle');
+      book.editField('title').name('new Title');
+    });
+    const plan = stripCallsites(migrationPlan(steps));
+    const contentTypes = [{
+      sys: { id: 'book' },
+      fields: [{
+        id: 'title'
+      }]
+    }];
+
+    const validationErrors = validatePlan(plan, contentTypes);
+    expect(validationErrors).to.eql([
+      {
+        type: 'InvalidAction',
+        message: 'You cannot edit the field "title" because its ID was changed to "newTitle" earlier in the migration.',
+        details: {
+          step: {
+            'type': 'field/update',
+            'meta': {
+              'contentTypeInstanceId': 'contentType/book/0',
+              'fieldInstanceId': 'fields/title/1'
+            },
+            'payload': {
+              'contentTypeId': 'book',
+              'fieldId': 'title',
+              'props': {
+                'name': 'new Title'
+              }
+            }
+          }
+        }
+      }
+    ]);
+  }));
+
+  it('does not return errors when switching ids for several fields and several times on the same field', Bluebird.coroutine(function * () {
```

Comment:
I think renaming the same field back & forth in the same migration shouldn't be possible, but renaming `title -> oldTitle` & `someOtherField -> title` in the same migration should work.